java/arcs/core/storage/StorageProxy.kt

```diff
@@ -330,6 +333,7 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperationAtTime, T>(
         check(stateHolder.value.state != ProxyState.NO_SYNC) {
             "getParticleView not valid on non-readable StorageProxy"
         }
+        checkInDispatcher()
```

Comment:
I think this one isn't needed, since the scheduler is used by this method (line 357)

Reply:
It's called directly from unit tests however. (StorageProxyTest)